### DISCUSSION TITLE
React 16 (Big project Webpack migration) Disable jsxInject

### DISCUSSION THREAD

**Discussion by mishabutuzov** [vitejs/vite-plugin-react#148] [Q&A]
Hi,

I want to use production code without any changes, but now - I need to remove all `import React from'react'`
How I can disable this (set skipReactImport to true), no options?

[link](https://github.com/vitejs/vite/blob/5d6ea8efc36bfdcd8b70afa8e82026ad1ccc0a77/packages/plugin-react/src/index.ts#L223)

![image](https://user-images.githubusercontent.com/20751556/232849152-08955b8e-e92e-4cd7-a1ff-14305140f725.png)

**Comment by ArnaudBarre** [comment]
What is your issue exactly? Even in case the check is not working, having an extra react import should not be an issue and should be removed by the bundler if not needed

**Comment by mishabutuzov** [comment]
I have duplicate react from'react' import error.

I use vite react latest typescript template, but change React and React Dom versions.
Downgrade to React 16. 

Thats my config:
`react({jsxRuntime: 'classic',}),`

**Reply by ArnaudBarre** [reply]
Is there any reason you're not using automatic runtime?

How does the import looks like in your source code?
Do you have anything else in your config that could alter jsx transformation?

**Reply by mishabutuzov** [reply]
automatic work perfect with React 17 and higher.
But on React 16 I get jsx-runtime error, so `jsxRuntime: 'classic'` was a solution.

**Reply by ArnaudBarre** [reply]
Are you on 16.14?

Anyway without more context I can't really help you. Maybe there is an issue with the regex that doesn't detect you current imports.
Bu my advice is to not waste time on this an instead try to use automatic runtime so that React import is not needed without any hack.
More info with links to codemod: https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

**Reply by mishabutuzov** [reply]
Thank you ArnaudBarre, Ill collect more information and post it.

I am using:

```
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-redux": "^8.0.2",
    "typescript": "4.5.5",
```

**Comment by ArnaudBarre** [comment]
For info this behaviour was dropped in v4
